src/main/java/sx/blah/discord/handle/impl/obj/Guild.java

```diff
@@ -48,13 +48,21 @@
 import java.time.LocalDateTime;
 import java.util.*;
 import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
 /**
  * The default implementation of {@link IGuild}.
  */
 public class Guild implements IGuild {
+
+	/**
+	 * This lock is utilized to prevent spam calls to {@link #setRegion(String)}.
+	 */
+	private final static Lock REGION_LOCK = new ReentrantLock();
```

Comment:
Why not use synchronized to be consistent with the source?

Reply:
As in the method have the synchronized keyword? Wouldn't work, since Guild instantiation is asynchronous an instance-level synchronize lock wouldn't have any effect.

Reply:
You could synchronize against the REGIONS field